pkg/lib/saml/service.go

```diff
@@ -563,10 +563,13 @@ func (s *Service) IssueLogoutResponse(
 func (s *Service) VerifyEmbeddedSignature(
 	sp *config.SAMLServiceProviderConfig,
 	samlElementXML string) error {
+	if !sp.ShouldVerifySignature {
+		return nil
+	}
 	certs, ok := s.SAMLSpSigningMaterials.Resolve(sp)
 	if !ok || len(certs.Certificates) == 0 {
-		// Signing cert not configured, nothing to verify
-		return nil
+		// This should be prevented by config validation. Therefore it is a programming error.
+		panic("SP certificates not configured but signature verification is required")
```

Comment:
[panic with error](https://github.com/authgear/authgear-server/issues/3626#:~:text=%C2%A0%234804%20(comment)-,panic%20with%20error,-Instead%20of%20panic). I will study whether there is golangci-lint rule to address this.

Reply:
Just curious, what is the advantage of panic with error instead of string?

Reply:
Updated that link to include this rationale.